src/components/carousel/index.js

```diff
@@ -261,7 +265,13 @@ export default class Carousel extends BaseComponent {
       update.initialOffset = {
         x: presenter.calcOffset(this.getThemeProps(), {currentPage: this.state.currentPage, pageWidth: containerWidth})
       };
+    } else {
+      update.pageWidth = pageWidth;
+      update.initialOffset = {
+        x: presenter.calcOffset(this.getThemeProps(), {currentPage: this.state.currentPage, pageWidth})
+      };
     }
+
```

Comment:
Is that necessary? What is it for?
The `onContainerLayout` was only meant for full-width pages
when the user passes his own `pageWidth` it's technically redundant. 
also in this case we already initializing the `state.pageWidth` in the constructor (with a little different value, btw).

Reply:
This fixes the offset and width when orientation changes (maybe this is why the `onOrientationChanged` code is redundant).
The width changes because the screen's width changes.
I'm not sure why the code in the constructor is that way (specifically the ` + this.getItemSpacings(themeProps)`), but I'm not sure if I should change it.

Reply:
but it's just because that in the example screen you decided to change pageWidth when orientation change, no? 
It doesn't necessarily relevant to the orientation change more than it's relevant to the pageWidth change. 
I think it will make more sense to handle this case when the user changes the pageWidth. WDYT?

Reply:
I see your point, but I tried to move the code to `UNSAFE_componentWillReceiveProps` and it actually had a bug when the initial page was set to something other than zero.
I'm thinking it might need both, or just the `onContainerLayout` if we want to handle just the orientation change and not actual width change. WDYT?

Reply:
Why using `UNSAFE_componentWillReceiveProps`? 
why not using `getDerivedStateFromProps` to save the new state values for `pageWidth` and `initialOffset`?

Reply:
Because that would require changing from `extends BaseComponent` to `asBaseComponent`, not sure I want to do that in this PR.

Reply:
ok.. acceptable.. 
So let's just add a TODO there..

Reply:
Done